patches/libuv/0001-Add-Managarm-support.patch

```diff
@@ -136,110 +194,192 @@ index 14d5f0c0..598b22c7 100644
 +  avg[0] = avg[1] = avg[2] = 0;
 +#endif
  }
-diff --git a/src/unix/linux-syscalls.c b/src/unix/linux-syscalls.c
-index 160056b4..7f20610d 100644
---- a/src/unix/linux-syscalls.c
-+++ b/src/unix/linux-syscalls.c
-@@ -22,7 +22,9 @@
- #include "linux-syscalls.h"
- #include <unistd.h>
- #include <signal.h>
-+#if !defined(__managarm__)
- #include <sys/syscall.h>
-+#endif
- #include <sys/types.h>
- #include <errno.h>
- 
-@@ -150,8 +152,10 @@ int uv__sendmmsg(int fd,
-                  struct uv__mmsghdr* mmsg,
-                  unsigned int vlen,
-                  unsigned int flags) {
--#if defined(__NR_sendmmsg)
-+#if defined(__NR_sendmmsg) && !defined(__managarm__)
-   return syscall(__NR_sendmmsg, fd, mmsg, vlen, flags);
-+#elif defined(__managarm__)
+diff --git a/src/unix/managarm-syscalls.c b/src/unix/managarm-syscalls.c
+new file mode 100644
+index 00000000..05af5755
+--- /dev/null
++++ b/src/unix/managarm-syscalls.c
+@@ -0,0 +1,93 @@
++/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
++ *
++ * Permission is hereby granted, free of charge, to any person obtaining a copy
++ * of this software and associated documentation files (the "Software"), to
++ * deal in the Software without restriction, including without limitation the
++ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
++ * sell copies of the Software, and to permit persons to whom the Software is
++ * furnished to do so, subject to the following conditions:
++ *
++ * The above copyright notice and this permission notice shall be included in
++ * all copies or substantial portions of the Software.
++ *
++ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
++ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
++ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
++ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
++ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
++ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
++ * IN THE SOFTWARE.
++ */
++
++#include "managarm-syscalls.h"
++#include <unistd.h>
++#include <signal.h>
++#include <sys/types.h>
++#include <errno.h>
++
++#if defined(__arm__)
++# if defined(__thumb__) || defined(__ARM_EABI__)
++#  define UV_SYSCALL_BASE 0
++# else
++#  define UV_SYSCALL_BASE 0x900000
++# endif
++#endif /* __arm__ */
```

Comment:
This is unnecessary? I don't see `UV_SYSCALL_BASE` referenced anywhere in the file